File: android/versioned-abis/expoview-abi30_0_0/src/main/java/abi30_0_0/host/exp/exponent/modules/internal/ExponentUnsignedAsyncStorageModule.java
// Copyright 2015-present 650 Industries. All rights reserved.

package abi30_0_0.host.exp.exponent.modules.internal;

import abi30_0_0.com.facebook.react.bridge.Callback;
import abi30_0_0.com.facebook.react.bridge.ReactApplicationContext;
import abi30_0_0.com.facebook.react.bridge.ReactContextBaseJavaModule;
import abi30_0_0.com.facebook.react.bridge.ReactMethod;
import abi30_0_0.com.facebook.react.bridge.ReadableArray;
import host.exp.exponent.kernel.KernelProvider;

public class ExponentUnsignedAsyncStorageModule extends ReactContextBaseJavaModule {

  private static final String ERROR_MESSAGE = "Can't use AsyncStorage in unsigned Experience.";

  public ExponentUnsignedAsyncStorageModule(ReactApplicationContext reactContext) {
    super(reactContext);
  }

  @Override
  public String getName() {
    return "AsyncSQLiteDBStorage";
  }

  @Override
  public boolean canOverrideExistingModule() {
    return true;
  }

  @ReactMethod
  public void multiGet(final ReadableArray keys, final Callback callback) {
    KernelProvider.getInstance().handleError(ERROR_MESSAGE);
  }

  @ReactMethod
  public void multiSet(final ReadableArray keyValueArray, final Callback callback) {
    KernelProvider.getInstance().handleError(ERROR_MESSAGE);
  }

  @ReactMethod
  public void multiRemove(final ReadableArray keys, final Callback callback) {
    KernelProvider.getInstance().handleError(ERROR_MESSAGE);
  }

  @ReactMethod
  public void multiMerge(final ReadableArray keyValueArray, final Callback callback) {
    KernelProvider.getInstance().handleError(ERROR_MESSAGE);
  }

  @ReactMethod
  public void clear(final Callback callback) {
    KernelProvider.getInstance().handleError(ERROR_MESSAGE);
  }

  @ReactMethod
  public void getAllKeys(final Callback callback) {
    KernelProvider.getInstance().handleError(ERROR_MESSAGE);
  }
}

File: android/versioned-abis/expoview-abi48_0_0/src/main/java/abi48_0_0/host/exp/exponent/modules/internal/ExponentUnsignedAsyncStorageModule.kt
// Copyright 2015-present 650 Industries. All rights reserved.
package abi48_0_0.host.exp.exponent.modules.internal

import abi48_0_0.com.facebook.react.bridge.*
import abi48_0_0.com.reactnativecommunity.asyncstorage.AsyncStorageModule
import host.exp.exponent.kernel.KernelProvider

class ExponentUnsignedAsyncStorageModule(reactContext: ReactApplicationContext) :
  ReactContextBaseJavaModule(reactContext) {
  override fun getName(): String {
    return AsyncStorageModule.NAME
  }

  override fun canOverrideExistingModule(): Boolean {
    return true
  }

  @ReactMethod
  fun multiGet(keys: ReadableArray?, callback: Callback?) {
    KernelProvider.instance.handleError(ERROR_MESSAGE)
  }

  @ReactMethod
  fun multiSet(keyValueArray: ReadableArray?, callback: Callback?) {
    KernelProvider.instance.handleError(ERROR_MESSAGE)
  }

  @ReactMethod
  fun multiRemove(keys: ReadableArray?, callback: Callback?) {
    KernelProvider.instance.handleError(ERROR_MESSAGE)
  }

  @ReactMethod
  fun multiMerge(keyValueArray: ReadableArray?, callback: Callback?) {
    KernelProvider.instance.handleError(ERROR_MESSAGE)
  }

  @ReactMethod
  fun clear(callback: Callback?) {
    KernelProvider.instance.handleError(ERROR_MESSAGE)
  }

  @ReactMethod
  fun getAllKeys(callback: Callback?) {
    KernelProvider.instance.handleError(ERROR_MESSAGE)
  }

  companion object {
    private const val ERROR_MESSAGE = "Can't use AsyncStorage in unsigned Experience."
  }
}
